### PR TITLE
feat: implement GIT_EDITOR support

### PR DESCRIPTION
**Pull request by jippi**
## Description

By setting the `GIT_EDITOR` environment variable to `meteor --as-git-editor` you will be able simply run `git commit <almost anything>`, where the only exception is `-m/--message`, and construct the commit message via the familiar `meteor` TUI.

The reason why `-m`/`--message` won't work is that `GIT_EDITOR` will not be run if the end-user already provides a message. It seems like a decent trade-off and would likely be what most folks would expect anyway.

Also implemented `--skip-intro` flag to use in conjunction with `--as-git-editor` allowing end-user to disable the intro via a flag, rather than repository configuration file (of which there might be none when using `GIT_EDITOR`)

## Testing

```shell
export GIT_EDITOR="meteor -e --skip-intro"
git commit --allow-empty
git commit <some file>
git commit -a
```

## Context

I just noticed this repo while looking at `huh` and thought it would be neat to use as a default `commit` interface, but didn't want to relearn or rewrite my tooling away from `git commit` behavior in general.

I'm fine with you not accepting it, was a fun quick hack that took ~30min, and I learned a bit about git while messing around :)

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -220,6 +223,7 @@ func main() {
 			huh.NewText().
 				Value(&newCommit.Body).
 				Title("Body").
+				CharLimit(0).
```
Comment (jippi) [inline main.go:226]: I added this because for some reason I hit a char limit while writing my commit to this PR, setting this explicitly to `0` fixed that

Comment (stefanlogue) [reply main.go:226]: Good catch! I hadn't run into that limit before

**Comment by jippi** [comment]
it's only downside right now is that `git commit --amend` is basically starting over every time

This could be fixed by parsing the file and putting in the values to the right vars in Go though; or detecting it and just opening the regular `$EDITOR` instead and bypass meteor entirely

**Comment by stefanlogue** [comment]
I think this belongs in a new issue, more than happy to accept a PR for the fix if you want to work on this, otherwise I'll get around to fixing it in the coming days/weeks.

**Comment by 5ouma** [comment]
> it's only downside right now is that `git commit --amend` is basically starting over every time
> 
> 
> 
> This could be fixed by parsing the file and putting in the values to the right vars in Go though; or detecting it and just opening the regular `$EDITOR` instead and bypass meteor entirely 

@jippi
Should I create a new Issue for this?

**Pull request merged**